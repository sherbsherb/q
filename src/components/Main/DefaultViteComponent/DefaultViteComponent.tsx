import { useState } from 'react'
import logo from './logo.svg'
import './App.css'
import styled, { useTheme } from 'styled-components'
import { theme } from '@/theme'

/**
 * Component with counter
 * @returns component with react spinner
 */

export function DefaultViteComponent(): JSX.Element {
  const [count, setCount] = useState(0)
  const theme = useTheme()
  console.log(theme)

  return (
    <div style={{ border: `1px solid ${theme.colors.red}` }}>
      <img
        src={logo}
        className="App-logo"
        alt="logo"
        width="300px"
        height="auto"
      />
      <p>
        <Button
          onClick={() => setCount((count) => count + 1)}
        >
          count is: {count}
        </Button>
      </p>
      <img src="img.jpg" alt="farmers and robots" />
    </div>
  )
}

const Button = styled.button`
  border-color:  ${props => props.theme.colors.red};
  background-color: ${theme.colors.grey};
  color: white;
  cursor: pointer;
`
