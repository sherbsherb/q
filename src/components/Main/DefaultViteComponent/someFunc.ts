export function someFunc(a: number, b: number): number {
  return a + b
}

someFunc(1, 2)
