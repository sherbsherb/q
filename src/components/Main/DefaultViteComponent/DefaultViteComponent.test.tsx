import { describe, expect, it, vi, afterEach, beforeEach } from 'vitest'
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import { DefaultViteComponent } from './DefaultViteComponent'
import '@testing-library/jest-dom'

// https://www.youtube.com/watch?v=oWJpxtAl62w

describe('Simple working test', () => {
  it('test library simply works', () => {
    expect(1 + 1).toEqual(2)
  })
  it('element should be visible', async () => {
    render(<DefaultViteComponent />)
    expect(screen.getByText('count is: 0')).toBeInTheDocument()
  })
  it('should increment', async () => {
    render(<DefaultViteComponent />)
    const button = screen.getByRole('button')
    fireEvent.click(button)
    await waitFor(() => expect(button).toHaveTextContent('count is: 1'))
  })
  it('toUpperCase', () => {
    const result = 'lower_case_string'.toUpperCase()
    expect(result).toMatchSnapshot()
  })
  it('toUpperCase', () => {
    const result = 'UPPER_CASE_STRING'.toLowerCase()
    expect(result).toMatchInlineSnapshot('"upper_case_string"')
  })
})
