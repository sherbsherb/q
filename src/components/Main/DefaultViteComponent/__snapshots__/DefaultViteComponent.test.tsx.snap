// Vitest Snapshot v1

exports[`Simple working test > toUpperCase 1`] = `"LOWER_CASE_STRING"`;
